### ISSUE TITLE
Feature request: Ask {leela, Ray, Aq} for next N moves

### ISSUE THREAD

**Issue by fishcu** [pnprog/goreviewpartner#10]
During review, I often like to ask the bot player repeatedly about the next move to see how the game would play out given a certain variation. To do that, I have to repeatedly click on the "ask move" button. It would be nice to have the possibility to generate a batch of bot moves at a time, for example 10 moves, or until the user presses "cancel" in a pop up window "generating continuation".

**Comment by pnprog** [comment]
I see. Something like "auto self play" starting from that position, with a way to pause it.
It's totally doable, the only annoyance is that once the bot start thinking, I cannot stop it until it came up with a move, so kind of annoying is the thinking time is very long.
But it's worth adding this option.

**Comment by fishcu** [comment]
Yes, I see that if there's no way to interrupt the bot, it could be a bad user experience. But I think being able to "cancel", then some feedback like "waiting for bot to finish this move..." should be OK!

**Comment by abhuva** [comment]
Just as an idea: How about having a seperate setting for thinking time for this kind of variation-review.  So i can have long thinking times during the review-process and faster ones during the variations.

**Comment by pnprog** [comment]
@abhuva 
Excellent idea!

**Comment by pnprog** [comment]
Hi @fishcu 

I just implemented the possibilities to have the bot reply as black, or white, or self playing during the review.

I think the result is quite good: the interface does not freeze anymore while waiting for the bot to come out with a move. Next, I will implement the idea from @abhuva

**Comment by pnprog** [comment]
Hi @abhuva ,
I just implemented different settings for analysis and review.

In the specific cases of Leela and Leela Zero, they also have a "time per move" parameter. It's possible to leave this parameter empty (or to set it to 0) and Leela(Z) will decide the thinking time by itself based on other parameters (for examples the number of playouts provided in the command line if provided).

**Comment by abhuva** [comment]
Nice, will love to test it once a compiled version is ready.

**Comment by pnprog** [comment]
@abhuva , v.10 has been released, you can download it from there: http://ns1297.imingo.net/
